code/modules/mob/living/carbon/carbon.dm

```diff
@@ -459,7 +459,7 @@
 	if((HAS_TRAIT(src, TRAIT_NOHUNGER) || HAS_TRAIT(src, TRAIT_TOXINLOVER)) && !force)
 		return TRUE
 
-	if(nutrition < 100 && !blood)
+	if(nutrition < 100 && !blood && !force)
```

Comment:
is there anything other than flyperson vomiting that uses a force argument of TRUE? it'd really be a shame if some other random source of vomiting became unable to make people dry heave because of this change.

Reply:
I reviewed every call on vomit, and nothing uses blood, and does not use force at the same time.
Abductor's come the closest without setting the force option, but the interaction is a positive I believe as it will not prevent the rejection of an implant by dry heaving when starving.

Reply:
The only issue would be something that sets `blood=false` and `force=true`.
This is the case for fly people, but nowhere else is force set to true without blood also being set to true as in the abductor code. Several places do set blood to true and that will still cause it to skip this block as intended.

Reply:
sounds good, then!